File: src/main/java/xyz/capybara/clamav/commands/scan/ContScan.java
package xyz.capybara.clamav.commands.scan;

import java.nio.ByteBuffer;
import java.nio.charset.StandardCharsets;

public class ContScan extends ScanCommand {

    public static final String COMMAND = "CONTSCAN";

    private String path;

    public ContScan(String path) {
        this.path = path;
    }

    @Override
    public String getCommandString() {
        return COMMAND;
    }

    @Override
    protected CommandFormat getFormat() {
        return CommandFormat.NEW_LINE;
    }

    @Override
    protected ByteBuffer getRawCommand() {
        StringBuilder rawCommand = new StringBuilder();
        rawCommand.append(getFormat().getPrefix())
                .append(getCommandString())
                .append(' ')
                .append(path)
                .append(getFormat().getTerminator());

        return ByteBuffer.wrap(rawCommand.toString().getBytes(StandardCharsets.US_ASCII));
    }
}

File: src/main/kotlin/xyz/capybara/clamav/commands/scan/ContScan.kt
package xyz.capybara.clamav.commands.scan

import java.nio.ByteBuffer
import java.nio.charset.StandardCharsets

internal class ContScan(private val path: String) : ScanCommand() {
    override val commandString
        get() = "CONTSCAN"

    override val format
        get() = CommandFormat.NEW_LINE

    override val rawCommand: ByteBuffer
        get() = ByteBuffer.wrap("${format.prefix}$commandString $path${format.terminator}".toByteArray(StandardCharsets.US_ASCII))
}
